back/spec/lib/participation_method/native_survey_spec.rb

```diff
@@ -175,13 +175,136 @@
   end
 
   describe '#user_fields_in_form?' do
-    it 'returns false when not enabled' do
+    it 'returns false if user_data_collection == \'anonymous\'' do
+      phase.permissions.find_by(action: 'posting_idea').update!(user_data_collection: 'anonymous')
       expect(participation_method.user_fields_in_form?).to be false
     end
 
-    it 'returns true when enabled' do
-      phase.user_fields_in_form = true
-      expect(participation_method.user_fields_in_form?).to be true
+    context 'when permission permitted_by is \'everyone\'' do
+      before do
+        permission = Permission.find_by(
+          permission_scope_id: phase.id,
+          action: 'posting_idea'
+        )
+
+        permission.permitted_by = 'everyone'
+        permission.save!
+      end
```

Comment:
Suggest changing this to a let, so you don't have to keep defining permission in every test (think this should work):

```suggestion
      let(:permission) { phase.permissions.find_by(action: 'posting_idea') }

      before do
        permission.update!(permitted_by: 'everyone')
      end
```